src/graphql/mirror.test.js

```diff
@@ -987,13 +989,30 @@ describe("graphql/mirror", () => {
         const spyFindOutdated = jest.spyOn(Mirror.prototype, "_findOutdated");
         const spyQueryFromPlan = jest.spyOn(Mirror.prototype, "_queryFromPlan");
         const spyCreateUpdate = jest.spyOn(Mirror.prototype, "_createUpdate");
-        const spies = [spyFindOutdated, spyQueryFromPlan, spyCreateUpdate];
+        const spyLogRequest = jest.spyOn(Mirror.prototype, "_logRequest");
+        const spyLogResponse = jest.spyOn(Mirror.prototype, "_logResponse");
+        const spyLogRequestUpdateId = jest.spyOn(
+          Mirror.prototype,
+          "_logRequestUpdateId"
+        );
+        const spies = [
+          spyFindOutdated,
+          spyQueryFromPlan,
+          spyCreateUpdate,
+          spyLogRequest,
+          spyLogResponse,
+          spyLogRequestUpdateId,
+        ];
 
         const postQuery = jest.fn();
         const now = jest
           .fn()
           .mockImplementationOnce(() => new Date(456))
-          .mockImplementationOnce(() => new Date(789));
+          .mockImplementationOnce(() => new Date(457))
+          .mockImplementationOnce(() => new Date(458))
+          .mockImplementationOnce(() => new Date(789))
+          .mockImplementationOnce(() => new Date(790))
+          .mockImplementationOnce(() => new Date(791));
```

Comment:
nit: there's a fair bit of coupling in this testing method
From a reviewers' pov, this amount of dates is difficult to keep in mind.

_Oh the joys of testing time! :sweat_smile:_

This seems to tread into "[solving puzzles, not problems](https://www.youtube.com/watch?v=2V1FtfBDsLU&t=4419)" territory. So I'm leaving this as a nit.

That said. The issue I see with coupling the test with implementation details this tightly, is that it becomes harder to change an implementation detail and know the program is still doing what it's supposed to. (i.e. the tests pass)

The easiest way to work around this I think, is to pass the `now: () => Date` function as an argument to the log functions, instead of the `Date` instances. This way, in a higher-level test, you don't need to call the function (only verify that it was passed).

Reply:
Do you think we need to update `_createUpdate` as well?

Reply:
I would go with the smallest PR. So leaving that as-is.
You could create a follow up for that if you like though.
Though do check with @decentralion / @wchargin, since they have more context on this.

Reply:
I'm going to leave this as-is unless we update the `_createUpdate` method as well.  I agree that the code you highlighted is onerous, but don't love the idea of using `options.now` on one line and `options.now()` a few lines later; the rationale of keeping the test code easier to read is kinda non-obvious if you were reading the module.